src/ai/defaultai.cc

```diff
@@ -1162,7 +1166,7 @@ void DefaultAI::late_initialization() {
 		}
 	}
 
-	// getting list of all fields nearby port space
+	// getting list of all fields nearby port space and a list of outer fields for shipyard priority
```

Comment:
all fields around portspaces in a radius of 3 are detected to be blocked for the ai to ensure we can build a port. 
all fields from radius 3 to radius 8 are marked to prefer shipyards.  
this should cure #1006 and is safeguarded by the check for wrong shipyard. good to test with europa 1.2 and black player